ThunderCats/UIColor+TCAdditions.m

```diff
@@ -28,12 +28,19 @@
 
 
 #import "UIColor+TCAdditions.h"
+#import "TCInvalidArgument.h"
 
 @implementation UIColor (TCAdditions)
 
 + (UIColor *)tc_colorWithHexString:(NSString *)hexString
 {
     NSString *cleanString = [hexString stringByReplacingOccurrencesOfString:@"#" withString:@""];
+    
+    if ([cleanString length] != 3 && [cleanString length] != 6)
+    {
+        [TCInvalidArgument raiseWithReason:@"invalid hex string"];
```

Comment:
This should be more like `[TCInvalidArgument raiseWithReason:@"%@ is not a valid hex color string.", hexString];`